rust/Cargo.lock

```diff
@@ -0,0 +1,204 @@
+[[package]]
```

Comment:
I think opinion is split on this somewhat, but I personally don't like shipping `Cargo.lock` in git.  Do pinning when we hit breakage (and ideally submit patches), rather than the constant maintenance overhead of bumping it.